lib/_stream_readable.js

```diff
@@ -798,7 +798,9 @@ Readable.prototype.pipe = function(dest, pipeOpts) {
     unpipe();
     dest.removeListener('error', onerror);
     if (EE.listenerCount(dest, 'error') === 0) {
-      if (!dest.destroyed) {
+      const s = dest._writableState || dest._readableState;
```

Comment:
The stream must have a `_writableState` in order to be writable and be a destination no?

Reply:
There is a weird test that fails without it. https://github.com/nodejs/node/blob/master/test/parallel/test-stream-auto-destroy.js#L89